src/wallet/account.ts

```diff
@@ -64,17 +66,11 @@ export class WalletAccount extends Account implements AccountInterface {
       this.channel.setChainId(res as StarknetChainId);
     });
 
-    // Get and Set Address !!! Post constructor initial empty string
-    walletProvider
-      .request({
-        type: 'wallet_requestAccounts',
-        params: {
-          silent_mode: false,
-        },
-      })
-      .then((res) => {
-        this.address = res[0].toLowerCase();
+    if (!address.length) {
+      requestAccounts(this.walletProvider).then(([accountAddress]) => {
+        this.address = accountAddress.toLowerCase();
       });
+    }
```

Comment:
Let's add a deprecation warning here saying that connect should be use instead.

Reply:
c83e304daa58d97b0e4abac6c06f03fb3ba1652e